inference-engine/ie_bridges/rust/crates/openvino/src/error.rs

```diff
@@ -0,0 +1,56 @@
+use thiserror::Error;
+
+/// See [IEStatusCode](https://docs.openvinotoolkit.org/latest/ie_c_api/ie__c__api_8h.html#a391683b1e8e26df8b58d7033edd9ee83).
+/// TODO Replace this in bindgen with [newtype_enum](https://docs.rs/bindgen/0.54.1/bindgen/struct.Builder.html#method.newtype_enum)
+/// or [rustified_enum](https://docs.rs/bindgen/0.54.1/bindgen/struct.Builder.html#method.rustified_enum).
+#[derive(Debug, Error)]
+pub enum InferenceError {
+    #[error("general error")]
+    GeneralError,
+    #[error("not implemented")]
+    NotImplemented,
+    #[error("network not loaded")]
+    NetworkNotLoaded,
+    #[error("parameter mismatch")]
+    ParameterMismatch,
+    #[error("not found")]
+    NotFound,
+    #[error("out of bounds")]
+    OutOfBounds,
+    #[error("unexpected")]
+    Unexpected,
+    #[error("request busy")]
+    RequestBusy,
+    #[error("result not ready")]
+    ResultNotReady,
+    #[error("not allocated")]
+    NotAllocated,
+    #[error("infer not started")]
+    InferNotStarted,
+    #[error("network not ready")]
+    NetworkNotReady,
+    #[error("undefined")]
+    Undefined,
```

Comment:
Maybe add the original `i32` as field to help with future debugging when new error codes get added to openvino, but not to openvino-rs.

Reply:
Good idea, done.